rust/otap-dataflow/crates/otap/src/otlp_exporter.rs

```diff
@@ -169,105 +235,101 @@ impl Exporter<OtapPdata> for OTLPExporter {
                     _ = metrics_reporter.report(&mut self.pdata_metrics);
                 }
                 Message::PData(pdata) => {
-                    // Capture signal type before moving pdata into try_from
-                    let signal_type = pdata.signal_type();
+                    if inflight_exports.len() >= max_in_flight {
+                        pending_msg = Some(Message::PData(pdata));
```

Comment:
Can we `debug_assert!(pending_msg.is_none())`?
I see it's from the `take()` above.

The control flow is somehow hard to follow. After the continue below we will enter a loop and the first if-statement is guaranteed true, which does a thing and then continues the loop. It leaves me wondering if the block of code at the top of the loop (`if let Some(completed) = inflight_exports.next_completion().await {`) couldn't just execute here, with one less `continue` statement maybe.

Reply:
I agree, I need to cleanup a bit the code here and better document.

Reply:
Added some clarifications